xds/src/main/java/io/grpc/xds/XdsClientWrapperForServerSds.java

```diff
@@ -84,7 +84,7 @@ public static XdsClientWrapperForServerSds newInstance(
     Bootstrapper.BootstrapInfo bootstrapInfo = bootstrapper.readBootstrap();
     final List<Bootstrapper.ServerInfo> serverList = bootstrapInfo.getServers();
     if (serverList.isEmpty()) {
-      throw new NoSuchElementException("No management server provided by bootstrap");
+      throw new FileNotFoundException("No management server provided by bootstrap");
```

Comment:
this does not look correct. it is not even an IO exception (this is why it is originally not an IO exception), consider using custom checked exception if there isn't any good exception to use.

Reply:
Yeah, I can create a custom checked exception or just return with a log message. I think the latter is preferred but let me know your preferrence.

Reply:
it is definitely preferred to throw exception especially if the caller need to do something.

Looked at the code i think the `getXdsClientWrapperForServerSds` shouldn't return `null`. it should just throw and  `serverProtocolNegotiator` can handle the exception (or just remove the `getXdsClientWrapperForServerSds` and inline it since it does not really belong to ServerSdsProtocolNegotiator); it sees exception so it can fallback seems more natural than hmm somehow we have null, lets fallback (not very idiomatic java, it is c style - no offense to c =p).